modules/packages/LinearAlgebra.chpl

```diff
@@ -763,16 +763,77 @@ proc _matmatMult(A: [?Adom] ?eltType, B: [?Bdom] eltType)
 {
   if Adom.rank != 2 || Bdom.rank != 2 then
     compilerError("Ranks are not 2 and 2");
+  if Adom.shape(1) != Bdom.shape(0) then
+    halt("Mismatched shape in matrix-matrix multiplication");
 
-  var C: [Adom.dim(0), Bdom.dim(1)] eltType;
-
-  // naive algorithm
-  forall (i,j) in C.domain do
-    C[i,j] = + reduce (A[i,..]*B[..,j]);
+  private use RangeChunk;
 
+  var C: [Adom.dim(0), Bdom.dim(1)] eltType;
+  if Adom.low == (0,0) && Bdom.low == (0,0) && 
```

Comment:
It could be nice to put these checks into a private helper function (I'm sure other functions will be using it down the road as well), so that this could read a little clearer, e.g.

```chpl
// I'm open to other names...
if hasDefaultIndices(A, B) { 
  _matmatMultHelper(A, B, C);
} else {
   ...
}
```